daemon/graphdriver/overlay2/check.go

```diff
@@ -40,23 +40,10 @@ func doesSupportNativeDiff(d string) error {
 	}()
 
 	// Make directories l1/d, l1/d1, l2/d, l3, work, merged
-	if err := os.MkdirAll(filepath.Join(td, "l1", "d"), 0o755); err != nil {
-		return err
-	}
-	if err := os.MkdirAll(filepath.Join(td, "l1", "d1"), 0o755); err != nil {
-		return err
-	}
-	if err := os.MkdirAll(filepath.Join(td, "l2", "d"), 0o755); err != nil {
-		return err
-	}
-	if err := os.Mkdir(filepath.Join(td, "l3"), 0o755); err != nil {
-		return err
-	}
-	if err := os.Mkdir(filepath.Join(td, workDirName), 0o755); err != nil {
-		return err
-	}
-	if err := os.Mkdir(filepath.Join(td, mergedDirName), 0o755); err != nil {
-		return err
+	for _, dir := range []string{"l1", "l2", "l3", workDirName, mergedDirName} {
+		if err := os.MkdirAll(filepath.Join(td, dir), 0o755); err != nil {
+			return err
+		}
 	}
```

Comment:
Thanks! So, while the new code is shorter, it also looks to be quite different from the code it replaces;

- old code did a `l1/d`, `l1/d1`, `l2/d`, `l3`, `work`, `merged`
- new code only create the top-level directories (`l1`, `l2`, `l3`)
- old code only used `MkdirAll` for the  `l1/d`, `l1/d1`, `l2/d` directories, and a more conservative / strict `Mkdir` for the other paths (which may, or may not be relevant w.r.t. either errors being handled and/or permissions on paths created (I didn't check that part))

Reply:
I apologize for my inattention. Fixed https://github.com/moby/moby/pull/47851/commits/853a6f50dd138e489e29a5242068e03a09c636d9